### ISSUE TITLE
asynchronous or blocking getters and setters

### ISSUE THREAD

**Issue by lovasoa** [locka99/opcua#166]
Hello,

   I cannot find in the documentation whether value getters and setters are supposed to be non-blocking or not. Looking at the code, it looks like they are indeed supposed to be. And there seem to be no way to define an asynchronous variable getter or setter.

   What is the recommended way to implement a getter or setter that needs to interact with the network or the filesystem ?

**Comment by locka99** [comment]
At present they're synchronous only. You could poll the current value and have your own thread in the background which refreshes that value.

**Comment by lovasoa** [comment]
That would be very inefficient !

**Comment by locka99** [comment]
Yes I agree. I think until async permeates through the api it would be the only way to do it.
